src/NuGetForUnity/Editor/PathHelper.cs

```diff
@@ -0,0 +1,223 @@
+﻿using System;
+using System.IO;
+using System.Runtime.CompilerServices;
+using System.Text;
+using UnityEngine;
+
+/// <summary>
+///     Helper class to handle path operations.
+/// </summary>
+internal static class PathHelper
+{
+    /// <summary>
+    ///     The path comparison type to use for the current platform the editor is running on.
+    /// </summary>
+    internal static readonly StringComparison PathComparisonType =
+        Application.platform == RuntimePlatform.WindowsEditor || Application.platform == RuntimePlatform.OSXEditor ?
+            StringComparison.OrdinalIgnoreCase :
+            StringComparison.Ordinal;
+
+    /// <summary>
+    ///     Create a relative path from one path to another. Paths will be resolved before calculating the difference.
+    ///     Default path comparison for the active platform will be used (OrdinalIgnoreCase for Windows or Mac, Ordinal for Unix).
+    /// </summary>
+    /// <remarks>
+    ///     Should the project upgrade to Unity version that has .NET Standard 2.1, this can be removed and instead we will use:
+    ///     Path.GetRelativePath(relativeTo, path).
+    ///     <see
+    ///         href="https://github.com/dotnet/runtime/blob/4aa77dfb0674fd61e41e574b8cc8ae63146f52c4/src/libraries/System.Private.CoreLib/src/System/IO/Path.cs#L843" />
+    ///     .
+    /// </remarks>
+    /// <param name="relativeTo">The source path the output should be relative to. This path is always considered to be a directory.</param>
+    /// <param name="path">The destination path.</param>
+    /// <returns>The relative path or <paramref name="path" /> if the paths don't share the same root.</returns>
+    internal static string GetRelativePath(string relativeTo, string path)
```

Comment:
Why do we need this complex function or, in other words, why isn't GetProjectRelativePath we implemented more than enough for this project? If you just wanted to have a polyfill for Path.GetRelativePath that we can just remove when we can use the real thing I don't see that happening since as I wrote above I don't see how we could ever use Path.GetRelativePath and have it work as we would like when run from CLI version. Unless we make CLI version call Directory.SetCurrentDirectory on start to change current directory to the path passed as argument, but that might be strange...

Reply:
The difference between the "Easy" implementation you had before and the complex implementation (the polyfill of `GetProjectRelativePath`) is the support of having a `path` that is outside the `relativeTo` path. So e.g. the project relative path of `Assets/../../' is '..'. General example `C:/Test/Packages` relative to  `C:/Test/Assets` is  `../Packages`.

Reply:
There are some feature requests like #527 that like to have the `package.config` and the `NuGet.config` outside the `Assets` directory. And to support this we need relative path support that is not only going "down" the directory tree.